src/Confluent.Kafka/Consumer.cs

```diff
@@ -671,77 +680,59 @@ public bool Consume(out Message message, int millisecondsTimeout)
             return false;
         }
 
+        /// <summary>
+        ///     Manually consume message or triggers events.
+        ///
+        ///     Will invoke events for OnPartitionsAssigned/Revoked,
+        ///     OnOffsetCommit, etc. on the calling thread.
+        /// </summary>
+        public bool Consume(out Message message, int millisecondsTimeout)
+        {
+            if (backgroundPollThread.IsStarted)
+            {
+                throw new Exception("Poll/Consume cannot be called on a Consumer when a background poll thread running.");
```

Comment:
Maybe a more specific exception - like `InvalidOperationException` - would be a better fit here?